timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -351,19 +350,17 @@ void declare_context(py::module &m) {
             "multiple_steps",
             [](Context &ctxt, const int n_steps, int store_x_interval) -> py::tuple {
                 // (ytz): I hate C++
-                int x_interval = (store_x_interval <= 0) ? n_steps : store_x_interval;
-                std::array<std::vector<double>, 2> result = ctxt.multiple_steps(n_steps, x_interval);
-
                 int N = ctxt.num_atoms();
                 int D = 3;
-                int F = result[0].size() / (N * D);
-                py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
-                std::memcpy(out_x_buffer.mutable_data(), result[0].data(), result[0].size() * sizeof(double));
-
-                py::array_t<double, py::array::c_style> box_buffer({F, D, D});
-                std::memcpy(box_buffer.mutable_data(), result[1].data(), result[1].size() * sizeof(double));
 
-                return py::make_tuple(out_x_buffer, box_buffer);
+                int x_interval = (store_x_interval <= 0) ? n_steps : store_x_interval;
+                int n_samples = store_x_interval > n_steps ? 0 : n_steps / x_interval;
+                py::array_t<double, py::array::c_style> out_x_buffer({n_samples, N, D});
+                py::array_t<double, py::array::c_style> box_buffer({n_samples, D, D});
+                auto res = py::make_tuple(out_x_buffer, box_buffer);
+                py::gil_scoped_release release;
```

Comment:
Question: is it possible/desirable to control whether or not we release the GIL from the Python API? This seems useful at least to strengthen `test_context_with_threads`, and might also be desirable to make releasing opt-in? (e.g. could be released in HREX; not sure if there are hazards in releasing by default)

Reply:
It is possible, but don't think it would be desirable. There should be no hazards in releasing the GIL, doing any IO releases the GIL (such as calling `time.sleep`). 

For the `test_context_with_threads` would triggering a failing test in CI be sufficient?

Reply:
Added a commit that should fail in https://github.com/proteneer/timemachine/pull/1370/commits/54144767596d948fefc0f948a9a73009ce6a84aa that simply remove the GIL release.